### PR TITLE
refactor(hr): update styled system implementation

### PR DESCRIPTION
**Pull request by mmaciorowski-sage**
BREAKING CHANGE: Use margin only. Padding has been removed.

### Proposed behaviour
`@styled-system/space` margin only (remove padding - breaking change).
Filter styled system margin props.

### Current behaviour
Uses no filtered `propTypes.space`.

### Checklist

- [x] Commits follow our style guide
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [ ] Carbon implementation and Design System documentation are congruent

### Testing instructions
- Padding props have been removed from story page props section.
- https://codesandbox.io/s/carbon-quickstart-forked-gmr2k?file=/src/index.js

### CODE REVIEW

**Review by dawidzarzycki — APPROVED** [review]

File: src/components/hr/hr.component.js [hunk 1]
```diff
@@ -27,8 +30,8 @@ const Hr = ({ adaptiveMxBreakpoint, ml, mr, ...props }) => {
 };
 
 Hr.propTypes = {
-  /** Styled system spacing props */
-  ...propTypes.space,
+  /** Filtered styled system margin props */
+  ...marginPropTypes,
```
Comment (dawidzarzycki) [inline src/components/hr/hr.component.js:34]: Is it not like we are changing from space into margins only in this case?

**Review by dawidzarzycki — CHANGES_REQUESTED** [review]
Left one comment, need confirmation

File: src/components/hr/hr.component.js (hunk 1 above)
Comment (dawidzarzycki) [reply src/components/hr/hr.component.js:34]: ignore this comment

**Review by dawidzarzycki — DISMISSED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by dawidzarzycki — DISMISSED** [review]

**Review by dawidzarzycki — DISMISSED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by KatarzynaQA — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by dawidzarzycki — APPROVED** [review]

**Review by KatarzynaQA — APPROVED** [review]

**Pull request merged**